bitcoin/src/blockdata/script/push_bytes.rs

```diff
@@ -84,12 +89,24 @@ mod primitive {
                     #[inline]
                     #[track_caller]
                     fn index(&self, index: $type) -> &Self::Output {
-                        // Slicing can not make slices longer
+                        // SAFETY: Slicing can not make slices longer.
                         unsafe {
                             Self::from_slice_unchecked(&self.0[index])
                         }
                     }
                 }
+
+                /// Script subslicing operation - read [slicing safety](#slicing-safety)!
```

Comment:
There's no such section in `PushBytes`, it's in `Script` but unrelated to this. I think this was caused by copying the macro, which also explains why there wasn't `IndexMut` already. (Also the comment about slicing safety should be removed from immutable impl.)

Reply:
I just blindly copied this because it has 3 slashes (doc comments) lol, will remove it

Reply:
> I think this was caused by copying the macro, which also explains why there wasn't `IndexMut` already.

I see the macro there does not have IndexMut, is there a reason why Script doesn't implement IndexMut?

Reply:
`Script` itself doesn't impl `IndexMut` because you can wildly distort a script by flipping individual bytes. For example changing the length of a datapush, which then changes how many subsequent bytes are interpreted as part of the push and which are interpreted as future opcodes, etc.,

Having said this, we have infallible conversions in both directions from `Vec<u8>` so it's not like we have invariants about script structure that we need to maintain. But our current API requires you convert to bare bytes before trying to modify a script, and this was a deliberate design choice.

Reply:
But `PushBytes` implementing `IndexMut` I think makes perfect sense.